flow/project.py

```diff
@@ -3313,18 +3308,26 @@ def select(operation):
                 # Change groups to available run _JobOperation(s)
                 with self._potentially_buffered():
                     operations = []
-                    for flow_group in flow_groups:
-                        for aggregate in self._get_aggregate_store(
-                            flow_group.name
-                        ).values():
-                            operations.extend(
-                                flow_group._create_run_job_operations(
-                                    self._entrypoint,
-                                    default_directives,
-                                    aggregate,
-                                    ignore_conditions,
+                    run_groups = set(self._gather_flow_groups(names))
+                    for (
+                        aggregate_store,
+                        aggregate_groups,
+                    ) in self._group_to_aggregator.inverse.items():
+                        matching_groups = set(aggregate_groups) & run_groups
```

Comment:
Do we have tests to check that the groups don't get executed unless the user specifically asked for them?

Reply:
I'm not entirely sure but I _think_ we have coverage of that. I'll defer this to a follow-up PR.

Reply:
I confirmed that tests fail on #416 if this check is not performed correctly (if more groups are run/status-fetched/submitted than were requested).